social_core/backends/orcid.py

```diff
@@ -76,7 +76,8 @@ def get_user_details(self, response):
 
             if name:
                 first_name = name.get("given-names", {}).get("value", "")
-                last_name = name.get("family-name", {}).get("value", "")
+                if name.get("family-name", None) is not None:
+                    last_name = name.get("family-name").get("value", "")
```

Comment:
```suggestion
                if (family_name := name.get("family-name", None)) is not None:
                    last_name = family_name.get("value", "")
```

Reply:
Sounds good, thanks!

I'll replace the existing commit with that change for a simpler commit history - hope that is what you would prefer to (let me know if not, I'm happy to change it back to two separate commits if that is your preference :) ) .

Reply:
Oh, actually, I think there is a feature to squash commits on merge in Github, perhaps? I'll just make it 2 commits for now, for easier review - still happy to squash it externally, of course (just let me know if you would like me to do so :) ) .

Reply:
Just confirming that all unit tests and my manual tests still pass the same as before with this change :+1: